gnovm/pkg/gnolang/coverage.go

```diff
@@ -0,0 +1,642 @@
+package gnolang
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"html/template"
+	"io/fs"
+	"math"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/gnolang/gno/tm2/pkg/commands"
+)
+
+// color scheme for coverage report
+const (
+	colorReset  = "\033[0m"
+	colorOrange = "\033[38;5;208m" // number of hits
+	colorRed    = "\033[31m"       // no hits
+	colorGreen  = "\033[32m"       // covered lines
+	colorYellow = "\033[33m"       // partial coverage, or executable but not executed lines
+	colorWhite  = "\033[37m"       // non-executable lines
+	boldText    = "\033[1m"
+)
+
+type (
+	fileCoverageMap map[string]FileCoverage
+	pathCache       map[string]string
+)
+
+// CoverageData stores code coverage information
+type CoverageData struct {
```

Comment:
Why are the types exported, if it's only used within the `gnolang` package?
Please see my comment about moving it into a separate lib